databricks/koalas/window.py

```diff
@@ -1104,7 +1095,12 @@ def count(self):
         2  2.0
         3  3.0
         """
-        return super(Expanding, self).count()
+        def count(scol):
+            return F.when(
+                F.row_number().over(self._unbounded_window) >= self._min_periods,
+                F.count(scol).over(self._window)
+            ).otherwise(F.lit(None))
+        return self._apply_as_series_or_frame(count).astype('float64')
```

Comment:
I might be missing something, but seems like pandas did the same thing for rolling as well?

https://github.com/pandas-dev/pandas/pull/30923/files#diff-0de5c5d9abfcdd141e83701eaaec4358L1194-L1195

What happens if we move this up to `_RollingAndExpanding`?

Reply:
@ueshin 

> What happens if we move this up to _RollingAndExpanding?

I tried, and it occurs test failure of `Rolling.count()` and `RollingGroupby.count()`.

and i think i found the reason like the below, which seems we might be fixing.

https://github.com/databricks/koalas/blob/f8b60d223c49da386b10e453ea2131db0d8623d2/databricks/koalas/generic.py#L1544-L1546

Reply:
i have tried, but it occurred the test failure for `Rolling.count()` and `RollingGroupby.count()` maybe because of below:

https://github.com/databricks/koalas/blob/f8b60d223c49da386b10e453ea2131db0d8623d2/databricks/koalas/window.py#L134-L139

i think we might be fixing this first.

Reply:
in short, the problem we have now like the below.

```python
>>> pser.rolling(5).count().sort_index()
0     1.0
1     2.0
2     3.0
3     4.0
4     5.0
5     5.0
6     5.0
7     5.0
8     5.0
9     5.0
10    5.0
dtype: float64

>>> kser.rolling(5).count().sort_index()
0     1.0
1     2.0
2     3.0
3     4.0
4     5.0
5     5.0
6     5.0
7     5.0
8     5.0
9     5.0
10    5.0
Name: 0, dtype: float64
```

it's okay, but.

```python
>>> pser.rolling(5, min_periods=3).count().sort_index()
0     NaN
1     NaN
2     3.0
3     4.0
4     5.0
5     5.0
6     5.0
7     5.0
8     5.0
9     5.0
10    5.0
dtype: float64

>>> kser.rolling(5, min_periods=3).count().sort_index()
0     1.0
1     2.0
2     3.0
3     4.0
4     5.0
5     5.0
6     5.0
7     5.0
8     5.0
9     5.0
10    5.0
Name: 0, dtype: float64
```

and if we do 

> What happens if we move this up to _RollingAndExpanding?

```python
>>> pser.rolling(5).count().sort_index()
0     1.0
1     2.0
2     3.0
3     4.0
4     5.0
5     5.0
6     5.0
7     5.0
8     5.0
9     5.0
10    5.0
dtype: float64

>>> kser.rolling(5).count().sort_index()
0     NaN
1     NaN
2     NaN
3     NaN
4     5.0
5     5.0
6     5.0
7     5.0
8     5.0
9     5.0
10    5.0
Name: 0, dtype: float64
```

since we're treating the `min_periods` is same with `window` when `min_periods` is not given. 

```python
kser.rolling(5).count() == kser.rolling(5, min_periods=5).count()
0     True
1     True
2     True
3     True
4     True
5     True
6     True
7     True
8     True
9     True
10    True
Name: 0, dtype: bool
```

we should fix it.

Reply:
Which error and test failure happens? It should be best to fix it together here.

Reply:
@HyukjinKwon 

Thanks for responding !

1. as i commented above, the parameter `min_periods`  is not working for `Rolling.count()` and `RollingGroupby.count()`



> ```python
> >>> pser.rolling(5, min_periods=3).count().sort_index()
> 0     NaN
> 1     NaN
> 2     3.0
> 3     4.0
> 4     5.0
> 5     5.0
> 6     5.0
> 7     5.0
> 8     5.0
> 9     5.0
> 10    5.0
> dtype: float64
> 
> >>> kser.rolling(5, min_periods=3).count().sort_index()
> 0     1.0
> 1     2.0
> 2     3.0
> 3     4.0
> 4     5.0
> 5     5.0
> 6     5.0
> 7     5.0
> 8     5.0
> 9     5.0
> 10    5.0
> Name: 0, dtype: float64
> ```

2. this is because the `count()` for `Rolling` and `RollingGroupby` is implemented like the below, without considering about `min_periods` at all

```python
    def count(self):
        def count(scol):
            return F.count(scol).over(self._window)

        return self._apply_as_series_or_frame(count).astype('float64')
```

3. so i have tried to fix this with applying same logic with `Expanding.count()` and `ExpandingGroupby.count()` like the below:

```python
        def count(scol):
            return F.when(
                F.row_number().over(self._unbounded_window) >= self._min_periods,
                F.count(scol).over(self._window)
            ).otherwise(F.lit(None))
        return self._apply_as_series_or_frame(count).astype('float64')
```

4. but it drive another bug looks like the below.

```python
>>> pser.rolling(5).count().sort_index()
0     1.0
1     2.0
2     3.0
3     4.0
4     5.0
5     5.0
6     5.0
7     5.0
8     5.0
9     5.0
10    5.0
dtype: float64

>>> kser.rolling(5).count().sort_index()
0     NaN
1     NaN
2     NaN
3     NaN
4     5.0
5     5.0
6     5.0
7     5.0
8     5.0
9     5.0
10    5.0
Name: 0, dtype: float64
```

since we're treating the `min_periods` as same as `window` if specific value is not given for `min_periods`

https://github.com/databricks/koalas/blob/f8b60d223c49da386b10e453ea2131db0d8623d2/databricks/koalas/window.py#L134-L139

Reply:
Okay, let's fix it separately.